docs-markdown/src/controllers/image-controller.ts

```diff
@@ -503,3 +512,44 @@ export function imageCompletionProvider() {
     completionItems.push(new CompletionItem(`:::image type="complex" source="" alt-text="" loc-scope="":::`));
     return completionItems;
 }
+
+export async function applyLink() {
+    // get editor, its needed to apply the output to editor window.
+    const editor = window.activeTextEditor;
+    if (!editor) {
+        noActiveEditorMessage();
+        return;
+    }
+
+    // if user has not selected any text, then continue
+    const RE_LINK = /:::image\s+((source|type|alt-text|lightbox|border|loc-scope)="([a-zA-Z0-9_.\/ -]+)"\s*)+:::/gm;
+    const position = new Position(editor.selection.active.line, editor.selection.active.character);
+    // get the current editor position and check if user is inside :::image::: tags
+    const wordRange = editor.document.getWordRangeAtPosition(position, RE_LINK);
+    if (wordRange) {
+        const options: InputBoxOptions = {
+            placeHolder: "Enter URL",
+            validateInput: (urlInput) => urlInput.startsWith("http://") || urlInput.startsWith("https://") ? "" :
+                "http:// or https:// is required for URLs. Link will not be added if prefix is not present.",
+        };
+        window.showInputBox(options).then((imageLink) => {
+            if (imageLink === undefined) {
+                postWarning("No URL given, abandoning command.");
+            } else {
+                editor.edit((selected) => {
+                    selected.insert(new Position(wordRange.end.line, wordRange.end.character - 3), ` link="${imageLink}"`);
+                });
+            }
+        });
+    } else {
+        const RE_LINK_EXISTS = /:::image\s+((source|type|alt-text|lightbox|border|loc-scope|link)="([a-zA-Z0-9_.\/ -:{1}]+)"\s*)+:::/gm;
```

Comment:
what is the :{1} for in this regex?

Reply:
This is used to look for a single colon in the url to support http(s)://. The link attribute wasn't captured with the current regex because the colon didn't match the pattern.

Reply:
Made the suggested regex change here and I'll create a work item to change the other regex patterns so we can test for regressions.